pkg/rke2/server.go

```diff
@@ -74,11 +77,23 @@ func setup(ctx *cli.Context, cfg Config) error {
 
 	managed.RegisterDriver(&etcd.ETCD{})
 
+	var cpConfig *podexecutor.CloudProviderConfig
+	if cfg.CloudProviderConfig != "" && cfg.CloudProviderName == "" {
+		return fmt.Errorf("cloud provider name has to be provided with cloud config path")
```

Comment:
The message here could be confusing, maybe: "--cloud-provider-config requires --cloud-provider-name"